### PR TITLE
Added method for convert uint8[32] into string

### LINKED ISSUE #5: Convert Sha256Hash to string
Hello,

I have necessity extract string at the value in the Sha256Has.

I have this code

```
string hello = "hello";
Bytes byte = asciiBytes(hello.c_str());
Sha256Hash shaHash = Sha256::getDoubleHash(byte.data(), byte.size());
```

And I have this test
`ASSERT_EQ(shaHash.value, "9595c9df90075148eb06860365df33584b75bff782a510c6cd4883a419833d50");`

Exists same mode for convert shaHash.value in the effective string, in this form "9595c9df90075148eb06860365df33584b75bff782a510c6cd4883a419833d50" ?

*Sorry for my terrible English but I'm leaning*

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
Hello,

I was have the necessity the hash converted into string and I added a method ToString()

I closing my issues open
- Closed #5 
- Closed: #6

### CODE REVIEW

**Comment by nayuki** [comment]
Hello Vincenzo, thanks for your attempt to contribute changes and describe your work.

Unfortunately, I cannot accept your contribution for the following reasons:

* My Bitcoin cryptography library is targeted toward embedded environments like Arduino. Because of this, I completely avoided heap memory allocation. You used the classes `std::stringstream` and `std::string`, which allocate memory on the heap, which is inappropriate for my design goals.
* You included credits and URLs as citation for a trivial number-to-hex-string algorithm. This is code that I can easily write myself and avoid the extra noise of unhelpful credits.
* Your comment block isn't formatted/aligned correctly. You left blank lines where there shouldn't be. You used a different brace style than my surrounding code.
* You needlessly altered whitespace in many places in my source file. I consider this an offensive, reckless change.
* You added logging prints to thecode (in Utils.cpp). You added a method from a different namespace `spyCBlock::UtilCrypto` which is completely inappropriate.
* In short, you made a contribution that solves your own problem in your own style, but does not help my library or my goals or fit the existing style of the code.

**Comment by vincenzopalazzo** [comment]
Hi @nayuki,

I didn't want to lack respect, I recognize my poor quality code.
Your library solved my problem and I think my pull request was constructive to improve my C ++ code, I'm sorry.

Yours is a great job.

**Comment by nayuki** [comment]
Thanks for your follow-up. The highest respect when contributing code is to make something that fits nicely with the existing project. This means paying great attention to style and avoiding unnecessary changes. Best wishes to your continual learning as a developer.

**Comment by vincenzopalazzo** [comment]
@nayuki Thanks :)
